test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -110,6 +110,16 @@ object Assertion extends AssertionVariants {
     final case class Function(name: String, paramLists: List[List[RenderParam]]) extends Render
     final case class Infix(left: RenderParam, op: String, right: RenderParam)    extends Render
 
+    /**
+     * Creates a string representation of a class name.
+     */
+    def className[A](C: ClassTag[A]): String =
+      try {
+        C.runtimeClass.getSimpleName
+      } catch {
+        case _: Throwable => C.runtimeClass.getName
```

Comment:
You could limit the catch to just `InternalError` ([as in the code @vasilmkd was referring to most likely](https://github.com/7mind/izumi/blob/2c7d27fb24fa467866aa08b3b9c3de239606b18a/fundamentals/fundamentals-reflection/src/main/scala/izumi/fundamentals/reflection/Tags.scala#L27))

Reply:
Good idea. Will do. Thank you!